yarn-project/end-to-end/src/e2e_l1_to_l2_msg.test.ts

```diff
@@ -175,4 +183,56 @@ describe('e2e_l1_to_l2_msg', () => {
     expect(consumptionReceipt.status).toBe(TxStatus.MINED);
     await expectBalance(ownerAddress, mintAmount + initialBalance - transferAmount);
   }, 80_000);
+
+  it('cancelled l1 to l2 messages cannot be consumed', async () => {
+    const initialBalance = 10n;
+    const [ownerAddress] = accounts;
+    const ownerPub = await aztecRpcServer.getAccountPublicKey(ownerAddress);
+    const deployedL2Contract = await deployContract(initialBalance, pointToPublicKey(ownerPub));
+    await expectBalance(accounts[0], initialBalance);
+
+    const l2TokenAddress = deployedL2Contract.address.toString() as `0x${string}`;
+
+    const { secretString, deadline } = await sendL1ToL2MessageViaPortal(ownerAddress, l2TokenAddress);
+    // Wait 5s for the archiver to process the message
+    await delay(5000);
+
+    // set the block timestamp to be after the deadline (so we can cancel the message)
+    const params = `["${deadline + 1}"]`;
+    await fetch(config.rpcUrl, {
+      body: `{"jsonrpc":"2.0", "method": "evm_setNextBlockTimestamp", "params": ${params}, "id": 1}`,
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+    });
+
+    // cancel the message
+    logger('cancelling the l1 to l2 message');
+    const args = [ownerAddress.toString(), 100n, deadline, secretString, 0n] as const;
+    await tokenPortal.write.cancelL1ToAztecMessage(args, { gas: 1_000_000n } as any);
+    // let archiver sync up
+    await delay(5000);
+
+    // archiver shouldn't have any pending messages.
```

Comment:
More a semantic, but these feel like an integration tests, and i think it should be moved into another file. This only covers testing the archiver instance and not the node as a whole.

Reply:
not sure I fully agree with this -> moving to another file would cause so much copy pasta since it requires so much of the setup used in this file